swift-sdk/Internal/InternalIterableAPI.swift

```diff
@@ -537,6 +547,12 @@ final class InternalIterableAPI: NSObject, PushTrackerProtocol, AuthProvider {
         }
     }
     
+    private func checkAndUpdateAuthToken(_ authToken: String? = nil) {
+        if config.authDelegate != nil && authToken != authManager.getAuthToken() {
+            onLogin(authToken)
+        }
+    }
+    
```

Comment:
AuthDelegate is a customer passed in handler that implements token-based auth right?